### PR TITLE
Setting up Zupoll TG Bot v1

### PR DESCRIPTION
**Pull request by vb7401**
Developing the necessary bot setup code, API endpoints, and frontend for the initial version of Zupoll TG Bot that:

- Sends a few introductory messages into a specific topic in the Zuzalu Social Hub supergroup
- Sends a new message every time a new poll is created

This is fully spec'd in https://0xparc.notion.site/2023-05-03-Vivek-Zupoll-TG-Bot-Spec-a05c2a4ad72e4ad6afb60bbaf33bbf61

### CODE REVIEW

File: apps/zupoll-server/src/util/bot.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    if (supergroup_id !== undefined && channel_id !== undefined) {
+      const message_thread_id = parseInt(channel_id);
+      const chat_id = parseInt(supergroup_id)
+      console.log("sending message to", chat_id, message_thread_id);
+      if (!isNaN(message_thread_id) && !isNaN(chat_id)) {
+        await bot.api.sendMessage(chat_id, message, {
+          message_thread_id,
+        });
+        // await bot.api.sendMessage("@vivboop", "Yes");
```
Comment (ichub) [inline apps/zupoll-server/src/util/bot.ts:16]: delete?

File: apps/zupoll-client/components/main/CreatePost.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import styled from "styled-components";
+import { ZupollError } from "../../src/types";
+
+export function CreatePost({
```
Comment (ichub) [inline apps/zupoll-client/components/main/CreatePost.tsx:4]: a little confused about this component, comment would be helpful

File: apps/zupoll-server/src/application.ts [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
-
+  let context: ApplicationContext = {};
+  const botToken = process.env.BOT_TOKEN;
+  if (botToken !== undefined) {
+    context = { 
+      bot: new Bot(botToken),
+    };
+    context.bot?.on("message", (ctx) => {
+      console.log(ctx);
+    });
```
Comment (ichub) [inline apps/zupoll-server/src/application.ts:17]: delete?

**Review by ichub — COMMENTED** [review]
My understanding of this code is that it's basically just testing out your ability to send stuff to telegram? Overall looks reasonable. Are you planning to get the stuff highlighted in red in your spec done today?

File: apps/zupoll-server/src/util/bot.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { Bot } from "grammy";
+
+export function cleanString(str: string) {
+  return str.replace("<", "&lt;").replace(">", "&gt;").replace("&", "&amp;");
+}
```
[the remark this answers is not in the document]
Comment (vb7401) [reply apps/zupoll-server/src/util/bot.ts:5]: These are the only ones that mess up the TG HTML parsing, so i just hardcoded. I put a link to the relevant TG spec though: https://core.telegram.org/bots/api#html-style

**Pull request merged**